File: python/sample01_05.py
# -*- coding: utf-8 -*-
#%% Sample 1-5
#%% 画像データの表現
# 画像の読み込みと表示 
# 
# 画像処理特論
# 
# 村松 正吾 
# 
# 動作確認: Python 3.7, PyTorch 1.8
#%% Digital image representation
# Image read and show
# 
# Advanced Topics in Image Processing
# 
# Shogo MURAMATSU
# 
# Verified: Python 3.7, PyTorch 1.8
from PIL import Image
import requests
import torchvision
from matplotlib import pyplot as plt 

#%% 画像の読み込み
# (Image read)
url = 'http://homepages.cae.wisc.edu/~ece533/images/peppers.png'
totensor = torchvision.transforms.ToTensor()
P = totensor(Image.open(requests.get(url, stream=True).raw))

#%% 画像の情報
# (Information of image)

print(P.dtype)
#%% 画像の表示
# (Image show)
topilimg = torchvision.transforms.ToPILImage()

plt.figure(1)
plt.imshow(topilimg(P))
plt.show()
#%% 
# © Copyright, Shogo MURAMATSU, All rights reserved.

File: scripts/sample01_05.m
%[text] # Sample 1-5
%[text] ## 画像データの表現
%[text] 画像の読み込みと表示 
%[text] 画像処理特論
%[text] 村松 正吾 
%[text] 動作確認: MATLAB R2023a
%[text] ## Digital image representation
%[text] Image read and show
%[text] Advanced Topics in Image Processing
%[text] Shogo MURAMATSU
%[text] Verified: MATLAB R2023a 
%%
%[text] ### 画像の読み込み
%[text]  (Image read)
P = imread('peppers.png');
%[text] ### 画像の情報
%[text]  (Information of image)
whos P
%%
%[text] ### 画像の表示
%[text]  (Image show)
figure(1)
imshow(P)
%%
%[text] © Copyright, Shogo MURAMATSU, All rights reserved.

%[appendix]{"version":"1.0"}
%---
%[metadata:view]
%   data: {"layout":"inline","rightPanelPercent":40}
%---
